src/tokens/wrappers/clawback/Clawback.sol

```diff
@@ -0,0 +1,308 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import {IClawback, IClawbackFunctions} from "./IClawback.sol";
+import {IERC721Transfer} from "../../common/IERC721Transfer.sol";
+
+import {IERC1155} from "@0xsequence/erc-1155/contracts/interfaces/IERC1155.sol";
+import {IERC165} from "@0xsequence/erc-1155/contracts/interfaces/IERC165.sol";
+import {ERC1155Metadata} from "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155Metadata.sol";
+import {ERC1155, ERC1155MintBurn} from "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155MintBurn.sol";
+
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+
+contract Clawback is ERC1155MintBurn, ERC1155Metadata, IClawback {
+    // Do not use address(0) as burn address due to common transfer restrictions.
+    address public constant BURN_ADDRESS = address(0x000000000000000000000000000000000000dEaD);
+
+    mapping(uint24 => Template) internal _templates;
+    mapping(uint256 => TokenDetails) internal _tokenDetails;
+
+    mapping(uint24 => mapping(address => bool)) public templateOperators;
+    mapping(uint24 => mapping(address => bool)) public templateTransferers;
+
+    uint24 private _nextTemplateId;
```

Comment:
Lets use `uint32` here and lets also pack it next to `_expectingReceive`, that way a single storage slot needs to be touch when creating a template

Reply:
`_expectingReceive` and `_nextTemplateId` are never used in the same function.

Reply:
Oh right, sorry about that